astropy/utils/decorators.py

```diff
@@ -230,8 +236,12 @@ def deprecated_attribute(name, since, message=None, alternative=None,
         user about this alternative if provided.
 
     pending : bool, optional
-        If True, uses a AstropyPendingDeprecationWarning instead of a
-        AstropyDeprecationWarning.
+        If True, uses a AstropyPendingDeprecationWarning instead of
```

Comment:
It does not feel right to allow a `warning_type` but somehow still enforces `AstropyPendingDeprecationWarning` when `pending=True`. I guess for LTS, we can live with this, but maybe open a new issue to revisit for 3.2?

Reply:
Yes, I wasn't sure what to do with that and had the same though that it can go in 3.2 if someone needs it. But then, `AstropyPendingDeprecationWarning` can be something astropy unique that cannot be customised further ;)

Reply:
It does seem a little silly to one and not the other, I am sure given some amount of time someone will want to customise the pending warning. It will probably be me lol.

Reply:
or me. please do open a follow-up PR, I'm swamped today.